controller/src/test/java/io/pravega/controller/server/ZKControllerServiceMainTest.java

```diff
@@ -59,11 +67,41 @@ public void tearDown() {
             Assert.fail("Error stopping test zk server");
         }
     }
+    
+    static class MockControllerServiceStarter extends ControllerServiceStarter {
```

Comment:
in the test you will see that after injecting sessionExpiration, we make a zookeeper call with a background callback supplied. and we make the MockServiceStarter to wait in its shutdown. we make the test wait on shutdown to complete. 

if curator never calls the callback then the test would timeout and we would know something bad is happening.

Note: The bug we identified was that if curator client is "closed" as there are background calls waiting for callback, curator does not invoke these callback. 
In controllerServiceMain, we were closing the curator client even before we had a chance to shutdown all our modules that depend on curator client (Note: their requests will still fail as session on curator client has expired. But they will fail and complete, not be hung forever because we closed the curator client).